doc/acl/Extend-L3V6ACLs.md

```diff
@@ -0,0 +1,428 @@
+  
+Using platform specific checks, AclOrch identifies platforms where  L3V4V6 ACL tables can be supported. Creation of an ACL table of type L3V4V6 on an unsupported platform results in orchagent logging an error in the syslog. A SAI ACL table create call is done only on supported platforms; this avoids crashes on unsupported platforms. In the future, orchagent would internally create separate v4 and v6 ACL tables on these unsupported platforms.
+
+##### STATE_DB
+A new field called `supported_L3V4V6` is added to the ACL capability in STATE_DB. This field is set to true by orchagent during 'AclOrch init' on platforms where v4 fields and v6 fields can be matched in the same hardware ACL table. This provides an interface to the operator to identify the platforms where the new L3V4V6 ACL table is supported. In the future, this field will be used by Orchagent to implement a single user configured L3V4V6 ACL table as two underlying SAI ACL tables on platforms where v4 fields and v6 fields cannot be matched in the same hardware ACL table.
+
+
+```
+127.0.0.1:6379[6]> hgetall "ACL_STAGE_CAPABILITY_TABLE|INGRESS"
+  :
+  :
+1) "supported_L3V4V6"
+2) "true"
+
+
+127.0.0.1:6379[6]> hgetall "ACL_STAGE_CAPABILITY_TABLE|EGRESS"
+  :
+  :
+5) "supported_L3V4V6"
+6) "true"
+```
+
+### SAI API 
+
+There are no new SAI APIs required for this feature.
+
+### Configuration and management 
+
+#### YANG model changes
+
+```
+--- a/models/yang/sonic/sonic-acl.yang
++++ b/models/yang/sonic/sonic-acl.yang
+
+    container sonic-acl {
+
+        container ACL_TABLE {
+                :
+                :
+                leaf type {
+                    type enumeration {
+                        enum MIRROR;
+                        enum MIRRORV6;
+                        enum L3;
+                        enum L3V6;
++                       enum L3V4V6;
+                    }
+                }
+
+                :
+                :
+            }
+        }
+  }
+
+
+```
+
+
+
+**Example:**
+```
+{
+    "ACL_TABLE": {
+        "DATAACL": {
+            "STAGE": "INGRESS",
+            "TYPE" : "L3V4V6",
+            "PORTS": [
+                "Ethernet0",
+                "Ethernet1"
+            ]
+        }
+    },
+    "ACL_RULE": {
+      "DATAACL|RULE1": {
+	    "ETHER_TYPE": "0x0800",
+	    "PRIORITY": "5",
+            "DST_IP": "20.2.2.2/32",
+	    "PACKET_ACTION": "DROP"
+        }
+      "DATAACL|RULE2": {
+	    "IP_TYPE": "IPV6ANY",
+	    "PRIORITY": "6",
+            "DST_IPV6": "2001::2/64",
+	    "PACKET_ACTION": "DROP"
+        }
+    }
+}
+```
+### CLI
+
+##### ACL table create CLI 
+The existing CLI is extended to support the new table type.
+```
+config acl add table -s <stage> -p <ports> <table_name> <table_type>
+```
+_table_type_ needs to be passed as _"L3V4V6"_ to create a table of the new L3V4V6 type.
+```
+Example : config acl add table -s ingress -p Ethernet0 DATAACL L3V4V6
+```
```

Comment:
We also need to update `acl-loader` script to support the new type.

Reply:
sonic-net/sonic-utilities#2794